src/lib/protocols/mdns/Publisher.h

```diff
@@ -0,0 +1,80 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "core/CHIPError.h"
+#include "inet/InetInterface.h"
+
+namespace chip {
+namespace Protocols {
+namespace Mdns {
+
+class Publisher
+{
+public:
+    Publisher() = default;
+
+    /**
+     * This method initializes the publisher.
+     *
+     * @retval CHIP_NO_ERROR                The initialization succeeds.
+     * @retval Error code                   The initialization fails.
+     *
+     */
+    CHIP_ERROR Init();
+
+    /**
+     * This method publishes the device on mDNS.
+     *
+     * This fucntion will fetch device name and other information and publish them
+     * via mDNS. If device meta data has changed, you can call this function again
+     * to update the information.
+     *
+     * @param[in] interface   The interface to send mDNS multicast.
+     *
+     * @retval CHIP_NO_ERROR                The publish succeeds.
+     * @retval CHIP_NO_ERROR                The publish succeeds.
```

Comment:
remove the duplicate return code.

Even better: remove all `@retval` items. It is obvious that NO_ERROR on success and error code on failure.